crmsh/utils.py

```diff
@@ -3038,4 +3032,34 @@ def get_systemd_timeout_start_in_sec(time_res):
     res_min = re.search("(\d+)min", time_res)
     start_timeout += 60 * int(res_min.group(1)) if res_min else 0
     return start_timeout
+
+
+def is_ocf_1_1_feature_on():
+    """
+    Only turn on ocf_1_1 feature when the cib schema version is pacemaker-3.7 or above
+    """
+    from .cibconfig import cib_factory
+    is_old_cib = schema.compare(cib_factory.get_schema(), constants.SCHEMA_MIN_VER_SUPPORT_OCF_1_1)
+    return not is_old_cib
+
+
+def handle_role_for_ocf_1_1(value, name='role'):
+    """
+    * Convert role from Promoted/Unpromoted to Master/Slave if schema doesn't support OCF 1.1
+    * Convert role from Master/Slave to Promoted/Unpromoted if schema support OCF 1.1
+    """
+    role_names = ["role", "target-role"]
```

Comment:
Yeah, this function is called in xmlutil.py and parse.py, covered all these cases:
```
target-role
role in monitor
role in location
role in colocaton
role in resource_set
role in rule
role in rsc_ticket
```

Reply:
```
crm(live/15sp3-1)configure# colocation col1 inf: d1:Promoted d2:Unpromoted
WARNING: Convert "Promoted" to "Master" since the current schema version is old and not upgraded yet. Please consider "cibadmin --upgrade --force"
WARNING: Convert "Unpromoted" to "Slave" since the current schema version is old and not upgraded yet. Please consider "cibadmin --upgrade --force"
```
Colocation role related code are handled and parsed by:
https://github.com/ClusterLabs/crmsh/blob/b683bf0da2c4d7167ae3cd50601b3e15b2e4a7df/crmsh/parse.py#L1028